vllm/worker/worker_base.py

```diff
@@ -81,3 +87,54 @@ def remove_lora(self, lora_id: int) -> bool:
 
     def list_loras(self) -> List[int]:
         raise ValueError(f"{type(self)} does not support LoRA")
+
+
+class WorkerWrapperBase:
+    """
+    The whole point of this class is to lazily initialize the worker.
+    We first instantiate the WorkerWrapper, which remembers the worker module
+    and class name. Then, when we call `update_environment_variables`, and the
+    real initialization happens in `init_worker`.
+    """
+
+    def __init__(self,
+                 init_cached_hf_modules=False,
+                 worker_module_name=None,
+                 worker_class_name=None) -> None:
+        self.worker_module_name = worker_module_name
+        self.worker_class_name = worker_class_name
+        self.init_cached_hf_modules = init_cached_hf_modules
+        self.worker = None
+
+    def update_environment_variables(self, envs: Dict[str, str]) -> None:
+        key = 'CUDA_VISIBLE_DEVICES'
+        if key in envs and key in os.environ:
+            # overwriting CUDA_VISIBLE_DEVICES is desired behavior
+            # suppress the warning in `update_environment_variables`
+            del os.environ[key]
+        update_environment_variables(envs)
+
+    def init_worker(self, *args, **kwargs):
```

Comment:
Can we add some docstrings?

Reply:
fixed in [21be004](https://github.com/vllm-project/vllm/pull/4024/commits/21be00449f892f6480b95e54d2cd94db1c46d720).

Reply:
This is a comment, docstrings look [like this](https://chat.openai.com/share/6cee29eb-f90b-4ae6-ac17-1464d24470bc) (this degree of formality is overkill for this method..)

Python doc tooling will then take the first statement (if it's a string) and expose it for auto doc generation

example:
https://github.com/vllm-project/vllm/blob/d150e4f89f9da5b093511227449ad940a9a82a52/vllm/core/block/cpu_gpu_block_allocator.py#L155-L167